extensions/warp-ipfs/src/store/message.rs

```diff
@@ -1632,6 +2504,156 @@ impl ConversationInner {
         Ok(())
     }
 }
+impl ConversationInner {
+    async fn create_community_task(&mut self, community_id: Uuid) -> Result<(), Error> {
+        let (ctx, crx) = mpsc::channel(256);
+
+        let task = community_task::CommunityTask::new(
+            community_id,
+            &self.ipfs,
+            &self.root,
+            &self.identity,
+            &self.file,
+            &self.discovery,
+            crx,
+            self.message_command.clone(),
+            self.event.clone(),
+        )
+        .await?;
+
+        let handle = self.executor.spawn_abortable(task.run());
+
+        tracing::info!(%community_id, "started community");
+
+        let inner_meta = CommunityInnerMeta {
+            command_tx: ctx,
+            _handle: handle,
+        };
+
+        self.community_task.insert(community_id, inner_meta);
+
+        Ok(())
+    }
+
+    async fn get_community_document(&self, id: Uuid) -> Result<CommunityDocument, Error> {
+        self.root.get_community_document(id).await
+    }
+
+    pub async fn set_community_document<B: BorrowMut<CommunityDocument>>(
+        &mut self,
+        mut document: B,
+    ) -> Result<(), Error> {
+        let document = document.borrow_mut();
+        let keypair = self.root.keypair();
+
+        let did = keypair.to_did()?;
+        if document.creator.eq(&did) {
+            document.sign(keypair)?;
+        }
+
+        document.verify()?;
+
+        self.root.set_community_document(document).await?;
+        self.identity.export_root_document().await?;
+        Ok(())
+    }
+    pub async fn contains_community(&self, id: Uuid) -> bool {
+        self.list_community_stream()
+            .await
+            .any(|community| async move { community.id() == id })
+            .await
+    }
+    pub async fn list_community(&self) -> Vec<CommunityDocument> {
+        self.list_community_stream().await.collect::<Vec<_>>().await
+    }
+    pub async fn list_community_stream(&self) -> impl Stream<Item = CommunityDocument> + Unpin {
+        self.root.list_community_document().await
+    }
+}
+
+impl ConversationInner {
+    pub async fn create_community(&mut self, mut name: &str) -> Result<Community, Error> {
```

Comment:
any reason for having `name` this as mutable vs doing `let name = name.trim();` ?